rust/rope/src/lib.rs

```diff
@@ -730,8 +779,9 @@ impl Node {
         (node.get_leaf(), offset)
     }
 
-    // precondition: offset > 0 and in range
     fn prev_codepoint_offset(&self, offset: usize) -> usize {
+        debug_assert!(offset > 0 && offset <= self.len());
```

Comment:
No, `<=` is valid, the return value should then be the index of the last codepoint in the buffer.